File: src/dataset/python/rotational-cipher/gemini/GeminiFlash_rotational-cipher_v2.py
def rotate(text, key):
    result = []
    for char in text:
        if 'a' <= char <= 'z':
            rotated_char = chr(((ord(char) - ord('a') + key) % 26) + ord('a'))
            result.append(rotated_char)
        elif 'A' <= char <= 'Z':
            rotated_char = chr(((ord(char) - ord('A') + key) % 26) + ord('A'))
            result.append(rotated_char)
        else:
            result.append(char)
    return ''.join(result)

File: src/dataset/c/rotational_cipher_Exercism_m3g4d1v3r/gemini/GeminiFlash_rotational_cipher_v2.c
#include "rotational_cipher.h"
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

char *rotate(const char *text, int shift_key) {
    if (text == NULL || *text == '\0') {
        return NULL;
    }

    size_t input_len = strlen(text);
    char *result = malloc(input_len + 1);

    if (result == NULL) {
        return NULL;
    }

    for (size_t i = 0; i < input_len; ++i) {
        char c = text[i];
        if (islower(c)) {
            result[i] = (char)(((c - 'a' + shift_key) % ALPHABET_SIZE + ALPHABET_SIZE) % ALPHABET_SIZE + 'a');
        } else if (isupper(c)) {
            result[i] = (char)(((c - 'A' + shift_key) % ALPHABET_SIZE + ALPHABET_SIZE) % ALPHABET_SIZE + 'A');
        } else {
            result[i] = c;
        }
    }

    result[input_len] = '\0';
    return result;
}